packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -193,7 +193,7 @@ export default function jsx({ settings, logging }: AstroPluginJSXOptions): Plugi
 
 			const { mode } = viteConfig;
 			// Shortcut: only use Astro renderer for MD and MDX files
-			if (id.includes('.mdx') || isMarkdownFile(id, { criteria: 'includes' })) {
+			if (id.endsWith('.mdx') || isMarkdownFile(id, { criteria: 'endsWith' })) {
```

Comment:
It also seems like with this change, `isMarkdownFile` will always be called with `criteria: 'endsWith'` everywhere else in the codebase. A follow-up PR in the future could be simplifying `isMarkdownFile`.

Reply:
I also noticed that and thought about it but I decided against it thinking of Chesterton's Fence.
@MoustaphaDev since you wrote that function, do you think the criteria param can be removed?

Reply:
@VladCuciureanu I wanted to point that out but I thought it would be a good candidate for a future PR, so yeah go ahead!

Reply:
Should be good now.